### PR TITLE
Health thread per host

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR runs a health thread per host. It's main purpose it so simplify a bit the health loop, and do the following:

1) Remove the health initialization from the host creation. This should speed up the latter.
2) Allow each host's health checks not to be delayed by another host's checks.
3) Skip on a host lock during the health loop.
4) We can have different health loop intervals per host.
5) When a child host is disconnected (but the host is in memory), nothing to run for it during the health loop.
6) Have a separate health.log file with related health logging.
7) Better functionality from aclk health (e.g. removed events now will be sent only for affected host).
8) Provide a base to further simplify/clean health (e.g. remove saving health log to files).

I'm not dead set on merging it if we don't think it's a very good idea in general, or if there is a blocker I haven't thought of. 

##### Test Plan

The basics should work, with children, notifications, up to cloud connectivity. Needs more testing e.g. on API, silencers, host labels alerts, chart labels in alerts, so this is still draft.

Since previously health was initiated from host, it was assumed things would be in place when charts and alerts are linked, so this is needed to be checked as well.

##### Additional Information

Sorry for the mess in `health.c`. In fact there is not so much change in the main health loop, but the diff is confused...

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by thiagoftsm** [comment]
Hello @MrZammler ,

Your PR is working as expected, I am waiting your answer about the small suggestion to approve PR.

LGTM!

**Comment by thiagoftsm** [comment]
I forgot to write, coverity scan did not report issues.

File: health/health.c [hunk 1]
```diff
@@ -593,15 +579,121 @@ static inline int check_if_resumed_from_suspension(void) {
...
+    char filename[FILENAME_MAX + 1];
+
+    if(!is_localhost) {
+        int r = mkdir(host->varlib_dir, 0775);
+        if (r != 0 && errno != EEXIST)
+            error("Host '%s': cannot create directory '%s'", rrdhost_hostname(host), host->varlib_dir);
+    }
+
+    {
+        snprintfz(filename, FILENAME_MAX, "%s/health", host->varlib_dir);
```
[the remark this answers is not in the document]
Comment (MrZammler) [reply health/health.c:636]: Thanks Thiago!

This, and all that have to do with the old health log file, should go, there's no reason to keep it around, most likely health log will end up in sqlite.

And in general the code in health needs to be refactored, the mail loop is many lines long and needs to be condensed, partly for readability.

For this PR I planned to do the least code changes needed just for the main thing to work to simplify as much as possible. Cleanup can be done afterwards.

**Review by thiagoftsm — APPROVED** [review]
Considering the tests result and the answer https://github.com/netdata/netdata/pull/13712#discussion_r995682771 received, LGTM!

File: database/rrd.h [hunk 2]
```diff
@@ -944,16 +944,18 @@ struct rrdhost {
...
-    STRING *health_default_exec;                    // the full path of the alarms notifications program
-    STRING *health_default_recipient;               // the default recipient for all alarms
-    char *health_log_filename;                      // the alarms event log filename
-    size_t health_log_entries_written;              // the number of alarm events written to the alarms event log
-    FILE *health_log_fp;                            // the FILE pointer to the open alarms event log file
-    uint32_t health_default_warn_repeat_every;      // the default value for the interval between repeating warning notifications
-    uint32_t health_default_crit_repeat_every;      // the default value for the interval between repeating critical notifications
-
+    unsigned int health_enabled;                   // 1 when this host has health enabled
+    volatile bool health_spawn;                    // true when health thread is running
```
Comment (vkalintiris) [inline database/rrd.h:948]: This should not be volatile.

**Review by thiagoftsm — APPROVED** [review]
It is working with child and parent on my environment, LGTM!

**Review by vkalintiris — APPROVED** [review]

**Pull request merged**